src/functions/routers/mod.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  ParameterSetDefinition,
+  PossibleParameterKeys,
+} from "../../parameters/mod.ts";
+import { SlackFunction } from "../mod.ts";
+import type { ActionHandler } from "./types.ts";
+
+/**
+ * Define an actions "router" and its input and output parameters for use in a Slack application. The ActionsRouter will route incoming action events to action-specific handlers.
+ * @param {SlackFunction<InputParameters, OutputParameters, RequiredInput, RequiredOutput>} func Reference to your previously-created SlackFunction, defined via DefineFunction
+ * @returns {ActionsRouter}
+ */
+export const BlockActionsRouter = <
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+>(
+  func: SlackFunction<
+    InputParameters,
+    OutputParameters,
+    RequiredInput,
+    RequiredOutput
+  >,
+) => {
+  return new ActionsRouter(func);
+};
+
+class ActionsRouter<
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+> {
+  private func: SlackFunction<
+    InputParameters,
+    OutputParameters,
+    RequiredInput,
+    RequiredOutput
+  >;
+  private routes: Array<
+    Array<string[] | RegExp | ActionHandler<typeof this.func.definition>>
+  >;
```

Comment:
How this is currently written will allow an array of any amount with no specific ordering. If I'm following correctly [this is intended to be a tuple](https://github.com/slackapi/deno-slack-sdk/pull/57/files#diff-72ea06050e8c947dc58ca4d49cd8fff07eeace6a35b0dd16b24ec1e5ab3b3828R72) where the first item can be an array of strings or a single RegExp.

If that's the case it seems like we could do
```ts
routes: [string[] | RegExp, ActionHandler<typeof this.func.definition>]
```

which will keep us to something like
```ts
this.routes = [['hi', 'fil' , 'how', 'are', 'you'], () => {}];
this.routes = [regExpValue, () => {}];
```

Reply:
Haha yeah good call, I struggled with how to type nested arrays with specific types at specific positions of the array 🤪 didn't know you could do what you suggest! Will give it a go.